service/router.go

```diff
@@ -76,62 +95,140 @@ func AdminRouter() *mux.Router {
+		Methods("GET")
+	router.Handle("/v1/accounts/upload", metric.CollectAPIStats("accountUpload",
+		MiddlewareWithCSRF(http.HandlerFunc(account.Upload)))).
+		Methods("POST")
+	router.Handle("/v1/accounts/{id:[0-9]+}/stores", metric.CollectAPIStats("substoreList",
+		MiddlewareWithCSRF(http.HandlerFunc(substore.List)))).
+		Methods("GET")
+	router.Handle("/v1/accounts/stores/{id:[0-9]+}", metric.CollectAPIStats("substoreUpdate",
+		MiddlewareWithCSRF(http.HandlerFunc(substore.Update)))).
+		Methods("PUT")
+	router.Handle("/v1/accounts/stores/{id:[0-9]+}", metric.CollectAPIStats("substoreDelete",
+		MiddlewareWithCSRF(http.HandlerFunc(substore.Delete)))).
+		Methods("DELETE")
+	router.Handle("/v1/accounts/stores", metric.CollectAPIStats("substoreCreate",
+		MiddlewareWithCSRF(http.HandlerFunc(substore.Create)))).
+		Methods("POST")
 
 	// API routes: system-user assertion
-	router.Handle("/v1/assertions", MiddlewareWithCSRF(http.HandlerFunc(assertion.SystemUserAssertion))).Methods("POST")
+	router.Handle("/v1/assertions", metric.CollectAPIStats("assertionSystemUserAssertion",
+		MiddlewareWithCSRF(http.HandlerFunc(assertion.SystemUserAssertion)))).
+		Methods("POST")
 
 	// API routes: users management
-	router.Handle("/v1/users", MiddlewareWithCSRF(http.HandlerFunc(user.List))).Methods("GET")
-	router.Handle("/v1/users", MiddlewareWithCSRF(http.HandlerFunc(user.Create))).Methods("POST")
-	router.Handle("/v1/users/{id:[0-9]+}", MiddlewareWithCSRF(http.HandlerFunc(user.Get))).Methods("GET")
-	router.Handle("/v1/users/{id:[0-9]+}", MiddlewareWithCSRF(http.HandlerFunc(user.Update))).Methods("PUT")
-	router.Handle("/v1/users/{id:[0-9]+}", MiddlewareWithCSRF(http.HandlerFunc(user.Delete))).Methods("DELETE")
-	router.Handle("/v1/users/{id:[0-9]+}/otheraccounts", MiddlewareWithCSRF(http.HandlerFunc(user.GetOtherAccounts))).Methods("GET")
+	router.Handle("/v1/users", metric.CollectAPIStats("XXX",
+		MiddlewareWithCSRF(http.HandlerFunc(user.List)))).
+		Methods("GET")
+	router.Handle("/v1/users", metric.CollectAPIStats("XXX",
+		MiddlewareWithCSRF(http.HandlerFunc(user.Create)))).
+		Methods("POST")
+	router.Handle("/v1/users/{id:[0-9]+}", metric.CollectAPIStats("XXX",
+		MiddlewareWithCSRF(http.HandlerFunc(user.Get)))).
+		Methods("GET")
+	router.Handle("/v1/users/{id:[0-9]+}", metric.CollectAPIStats("XXX",
+		MiddlewareWithCSRF(http.HandlerFunc(user.Update)))).
+		Methods("PUT")
+	router.Handle("/v1/users/{id:[0-9]+}", metric.CollectAPIStats("XXX",
+		MiddlewareWithCSRF(http.HandlerFunc(user.Delete)))).
+		Methods("DELETE")
+	router.Handle("/v1/users/{id:[0-9]+}/otheraccounts", metric.CollectAPIStats("XXX",
+		MiddlewareWithCSRF(http.HandlerFunc(user.GetOtherAccounts)))).
+		Methods("GET")
 
 	// OpenID routes: using Ubuntu SSO
-	router.Handle("/login", MiddlewareWithCSRF(http.HandlerFunc(usso.LoginHandler)))
-	router.Handle("/logout", MiddlewareWithCSRF(http.HandlerFunc(usso.LogoutHandler)))
+	router.Handle("/login", metric.CollectAPIStats("XXX",
```

Comment:
Either you've missed something here with the `XXX` or it could be clearer that this is intentional please!

Reply:
Omg, it was the placeholder I missed, fixing